cadquery/occ_impl/shapes.py

```diff
@@ -1322,6 +1322,17 @@ def makeLine(cls: Type["Edge"], v1: Vector, v2: Vector) -> "Edge":
         """
         return cls(BRepBuilderAPI_MakeEdge(v1.toPnt(), v2.toPnt()).Edge())
 
+    def radius(self) -> float:
```

Comment:
Could you move this method to the Mixin1D? I think it'll be then automatically usable for wires with one edge.

Reply:
`.geomType()` of a Wire always seems to be `"Wire"`, so as it is currently written it won't work for a wire with a single `"CIRCLE"` edge.

I could always add the following though:
```python
class Wire():
    ...
    def radius(self):
        radii = [el.radius() for el in self.Edges()]
        if not all(radii):
            raise ValueError('Not a circular wire')
        # calculate average
        # if deviation is above tolerance
            raise ValueError('Wire does not have a consistent radius')
        return avg_radius
```

Reply:
True, though you can still call `_geomAdaptor().Circle().Radius()` on a wire with one edge. Maybe catch `Standard_NoSuchObject` and rethrow it as a ValueError instead of checking `geomType`?

Reply:
I've been avoiding doing that because I get segfaults with certain non-circular edges. Can you recreate this segfault?

```
>>> w0 = cq.Wire.makeEllipse(20, 10, (0, 0, 0), (0, 0, 1), (1, 0, 0))
>>> w0._geomAdaptor().Circle()
zsh: segmentation fault (core dumped)  python
```

The only method I could think of to prevent this segfault is that `geomType() == "CIRCLE"` check. Which sucks, because it's pretty cool that OCCT can create a circle from anything that resembles a circle.

I am still on OCCT 7.4.0 (I'm not using Conda, but I will get the update done soon, especially if it's relevant to this PR).

Reply:
I get this with OCCT 7.4:
```python
Standard_NoSuchObject: GeomAdaptor_Curve::Circle() - curve is not a Circle
```

Reply:
Oh damn, I must have messed something up in my install. I'll move to a VM and Conda for the rest of this PR. Thanks for testing that for me.

I'll move the method over to Mixin1D and raise a ValueError from the OCP exception.